### ISSUE TITLE
Change Request: Allow the same rule with different severty

### ISSUE THREAD

**Issue by a7madgamal** [eslint/eslint#16040]
### ESLint version

8.17.0

### What problem do you want to solve?

The level of customisation in eslint is amazing and is the reason why it won the linters world cup :D
However, this opens the door for a feature that will take this to the next level: allowing the same rule to be defined with different severty levels based on different configs.
Sometimes I want to define a rule with a certain config as a warning and the same rule with another config as error

### What do you think is the correct solution?

For example, I want to be able to do this inside the rules object:
```js
{
  rules: {
    "@typescript-eslint/ban-types": [
      "warn",
      {
        types: {
          unknown: "unknown is bad, but a warning is ok 🤷🏻",
        },
      },
    ],
    "@typescript-eslint/ban-types": [
      "error",
      {
        types: {
          any: "any is really bad, this needs to be an error! ❌",
        },
      },
    ],
  },
}
```

I realize the following:
- in that specific example we might argue that this is a feature request for `@typescript-eslint/ban-types`. It's true, but I faced this many times before and will keep facing it with other rules. 
Allowing this in eslint itself will open a whole new door for ALL eslint rules 🤯 
- the keys are duplicate, this can be easily fixed by allowing rules to be an array instead of an object OR defining a way to scope rules like `"@typescript-eslint/ban-types:case1" ,"@typescript-eslint/ban-types#case1"` and showing this scope in errors and warnings

### Participation

- [X] I am willing to submit a pull request for this change.
